File: InventoryManagement.Core/Entities/Customer.cs
using InventoryManagement.Core.Common;

namespace InventoryManagement.Core.Entities;

public class Customer : BaseEntity
{
    public string CustomerName { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string? PhoneNumber { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public string? Country { get; set; }
    public string? PostalCode { get; set; }

    // Navigation Properties
    public ICollection<SalesOrder> SalesOrders { get; set; } = new List<SalesOrder>();
    public ICollection<Quotation> Quotations { get; set; } = new List<Quotation>();
}

File: InventoryManagement.UI/src/app/core/models/customer.ts
export interface Customer {
  id: number;
  customerName: string;
  email?: string;
  phoneNumber?: string;
  address?: string;
  city?: string;
  country?: string;
  postalCode?: string;
  createdAt: Date;
}

export interface CreateCustomerRequest {
  customerName: string;
  email?: string;
  phoneNumber?: string;
  address?: string;
  city?: string;
  country?: string;
  postalCode?: string;
}

export interface UpdateCustomerRequest {
  customerName: string;
  email?: string;
  phoneNumber?: string;
  address?: string;
  city?: string;
  country?: string;
  postalCode?: string;
}